offer-manager-domain/src/main/groovy/life/qbic/business/offers/update/UpdateOffer.groovy

```diff
@@ -44,22 +64,69 @@ class UpdateOffer implements UpdateOfferInput{
                 .identifier(identifier)
                 .build()
 
-        storeOffer(finalizedOffer)*/
+        if(oldOffer.checksum() != finalizedOffer.checksum()){
+            storeOffer(finalizedOffer)
+        }else{
+            output.failNotification("An unchanged offer cannot be updated")
+        }
     }
 
+
     private void storeOffer(Offer finalizedOffer) {
         try {
             final offer = Converter.convertOfferToDTO(finalizedOffer)
             dataSource.store(offer)
-            output.updatedOffer(offer)
+            output.createdNewOffer(offer)
         } catch (DatabaseQueryException e) {
             output.failNotification(e.message)
         } catch (Exception unexpected) {
-            //TODO use logger facade instead of println
-            println unexpected.message
-            println unexpected.stackTrace.join("\n")
+            log.error unexpected.message
+            log.error unexpected.stackTrace.join("\n")
             output.failNotification("An unexpected during the saving of your offer occurred. " +
                     "Please contact ${Constants.QBIC_HELPDESK_EMAIL}.")
         }
     }
+
+    private OfferId increaseOfferIdentifier(life.qbic.datamodel.dtos.business.OfferId oldOfferId){
+        OfferId convertedId = null
+
+        try{
+            //search for all ids in the database
+            List<life.qbic.datamodel.dtos.business.OfferId> allVersionIds = dataSource.fetchAllVersionsForOfferId(oldOfferId.toString())
```

Comment:
I also stumbled over that. I decided to implement it as is and wait for the reviewer to criticize it. Since you did I will update the method as you suggested.

Reply:
I thought about it again and would opt for keeping it as is. Searching for similar versions does not require a lot of logic and based on the ID the random and the project part can be easily obtained. Though I changed the type from string to offerID to be more explicit and use the id DTO. (ac37618)